app/javascript/components/time_table/timesheet/entry.js

```diff
@@ -121,16 +123,29 @@ class Entry extends React.Component {
       }).then((response) => {
           if (response.data.id) {
             this.props.pushEntry(response.data);
-            this.setState({
-              starts_at: this.state.ends_at,
-              duration: 0,
-              durationHours: '00:00',
+            const newState = {
               body: '',
               task: ''
-            })
+            };
+            if (!this.state.project.autofill) {
+              Object.assign(newState, { starts_at: this.state.ends_at, duration: 0, durationHours: '00:00' })
```

Comment:
You can use split instead of assign
`newState = { ...newState, starts_at: this.state.ends_at, duration: 0, durationHours: '00:00'  }`

Reply:
`Object.assign` does not introduce a reassignment, so `newState` can still be a const.